provisioning_client/adapters/hsm_client_riot.c

```diff
@@ -211,21 +198,23 @@ static int generate_root_ca_info(HSM_CLIENT_X509_INFO* riot_info, RIOT_ECC_SIGNA
             result = 0;
         }
     }
+    ecc_signature_destroy(&tbs_sig);
     return result;
 }
 
-static int produce_device_cert(HSM_CLIENT_X509_INFO* riot_info, RIOT_ECC_SIGNATURE tbs_sig, CERTIFICATE_SIGNING_TYPE signing_type)
+static int produce_device_cert(HSM_CLIENT_X509_INFO* riot_info, CERTIFICATE_SIGNING_TYPE signing_type)
 {
     int result;
     uint8_t der_buffer[DER_MAX_TBS] = { 0 };
     DERBuilderContext der_ctx = { 0 };
     RIOT_STATUS status;
+    RIOT_ECC_SIGNATURE tbs_sig = { 0 };
 
     if (signing_type == type_self_sign)
     {
         // Build the TBS (to be signed) region of DeviceID Certificate
         DERInitContext(&der_ctx, der_buffer, DER_MAX_TBS);
-        if (X509GetDeviceCertTBS(&der_ctx, &X509_DEVICE_TBS_DATA, &riot_info->device_id_pub) != 0)
+        if (X509GetDeviceCertTBS(&der_ctx, &X509_ROOT_TBS_DATA, &riot_info->ca_root_pub, (uint8_t*)&riot_info->ca_root_pub, sizeof(riot_info->ca_root_pub)) != 0)
```

Comment:
It looks like `car_root_pub` is of type `RIOT_ECC_PUBLIC` which resolves in the following form
`RIOT_ECC_PUBLIC` -> `ecc_publickey` -> `affine_point_t` ->
```c
typedef struct {
    bigval_t x;
    bigval_t y;
    uint32_t infinity;
} affine_point_t;
```

Just double checking that this should be casted to a `uint8_t*`?

Reply:
It should be RIOT_ECC_PUBLIC->mbedtls_ecp_point-> typedef struct mbedtls_ecp_point
{
    mbedtls_mpi X;          /*!< The X coordinate of the ECP point. */
    mbedtls_mpi Y;          /*!< The Y coordinate of the ECP point. */
    mbedtls_mpi Z;          /*!< The Z coordinate of the ECP point. */
}
mbedtls_ecp_point;
RIOT_ECC_PRIVATE->
typedef struct mbedtls_mpi
{
    int s;              /*!<  integer sign      */
    size_t n;           /*!<  total # of limbs  */
    mbedtls_mpi_uint *p;          /*!<  pointer to limbs  */
}
mbedtls_mpi;

Reply:
However I do notice that &X509_ROOT_TBS_DATA should be &X509_DEVICE_TBS_DATA, this was one of the API changes the final two parameters are optional: 
X509GetDeviceCertTBS(
    DERBuilderContext   *Tbs,
    RIOT_X509_TBS_DATA  *TbsData,
    RIOT_ECC_PUBLIC     *DevIdKeyPub,
    uint8_t             *RootKeyPub,
    uint32_t             RootKeyPubLen)
I also notice ca_root_pub should be device_id_pub: I suspect this will be the same mistake for the alias cert as well: Good find!

Reply:
Fixed